src/WinRT.Runtime/ComWrappersSupport.net5.cs

```diff
@@ -161,7 +166,217 @@ public static void InitializeComWrappers(ComWrappers wrappers = null)
+                    // The pointer to dispatch on for the instance.
+                    // ** Never release.
+                    classNative.Release |= ReleaseFlags.None; // Instance
+
+                    // A pointer to the inner that should be queried for
+                    //    additional interfaces. Immediately after a QueryInterface()
+                    //    a Release() should be called on the returned pointer but the
+                    //    pointer can be retained and used.
+                    // ** Release in this class's Finalizer.
+                    classNative.Release |= ReleaseFlags.Inner; // Inner
+                }
+                else
+                {
+                    // WinUI scenario
+                    // The pointer to dispatch on for the instance.
+                    // ** Never release.
+                    classNative.Release |= ReleaseFlags.None; // Instance
+
+                    // A pointer to the inner that should be queried for
+                    //    additional interfaces. Immediately after a QueryInterface()
+                    //    a Release() should be called on the returned pointer but the
+                    //    pointer can be retained and used.
+                    // ** Never release.
+                    classNative.Release |= ReleaseFlags.None; // Inner
+
+                    // No longer needed.
+                    // ** Never release.
+                    classNative.Release |= ReleaseFlags.None; // ReferenceTracker
+                }
+            }
+            else
+            {
+                if (classNative.ReferenceTracker == default(IntPtr))
+                {
+                    // COM scenario
+                    // The pointer to dispatch on for the instance.
+                    // ** Release in this class's Finalizer.
+                    classNative.Release |= ReleaseFlags.Instance; // Instance
+                }
+                else
+                {
+                    // WinUI scenario
+                    // The pointer to dispatch on for the instance.
+                    // ** Release in this class's Finalizer.
+                    classNative.Release |= ReleaseFlags.Instance; // Instance
+
+                    // This instance should be used to tell the
+                    //    Reference Tracker runtime whenever an AddRef()/Release()
+                    //    is performed on newInstance.
+                    // ** Release in this class's Finalizer.
+                    classNative.Release |= ReleaseFlags.ReferenceTracker; // ReferenceTracker
+                }
+            }
+
+            if (isAggregation)
+            {
+                Marshal.AddRef(newInstance);
+                var outerRC = Marshal.Release(newInstance);
+                if (outerRC != 0)
+                {
+                    // In aggregation scenarios, at this point outer's refcount must be 0
+                    System.Diagnostics.Debugger.Break();

```

Comment:
I thought this will crash a process if a debugger is not attached. Is that desirable? Perhaps check with `Debugger.IsAttached`? Alternatively if a debugger isn't attached throw an exception?

Reply:
thanks - will wrap with IsAttached (as well as BreakOnDispose, if I decide to check that it)